cropper/src/main/java/com/canhub/cropper/BitmapUtils.kt

```diff
@@ -0,0 +1,933 @@
+        get() {
+            // Safe minimum default size
+            val IMAGE_MAX_BITMAP_DIMENSION = 2048
+            return try {
+                // Get EGL Display
+                val egl = EGLContext.getEGL() as EGL10
+                val display = egl.eglGetDisplay(EGL10.EGL_DEFAULT_DISPLAY)
+                // Initialise
+                val version = IntArray(2)
+                egl.eglInitialize(display, version)
+                // Query total number of configurations
+                val totalConfigurations = IntArray(1)
+                egl.eglGetConfigs(display, null, 0, totalConfigurations)
+                // Query actual list configurations
+                val configurationsList = arrayOfNulls<EGLConfig>(
+                    totalConfigurations[0]
+                )
+                egl.eglGetConfigs(
+                    display,
+                    configurationsList,
+                    totalConfigurations[0],
+                    totalConfigurations
+                )
+                val textureSize = IntArray(1)
+                var maximumTextureSize = 0
+                // Iterate through all the configurations to located the maximum texture size
+                for (i in 0 until totalConfigurations[0]) {
+                    // Only need to check for width since opengl textures are always squared
+                    egl.eglGetConfigAttrib(
+                        display, configurationsList[i], EGL10.EGL_MAX_PBUFFER_WIDTH, textureSize
+                    )
+                    // Keep track of the maximum texture size
+                    if (maximumTextureSize < textureSize[0]) {
+                        maximumTextureSize = textureSize[0]
+                    }
+                }
+                // Release
+                egl.eglTerminate(display)
+                // Return largest texture size found, or default
+                Math.max(maximumTextureSize, IMAGE_MAX_BITMAP_DIMENSION)
+            } catch (e: Exception) {
+                IMAGE_MAX_BITMAP_DIMENSION
+            }
+        }
+
+    /**
+     * Close the given closeable object (Stream) in a safe way: check if it is null and catch-log
+     * exception thrown.
+     *
+     * @param closeable the closable object to close
+     */
+    private fun closeSafe(closeable: Closeable?) {
+        if (closeable != null) {
+            try {
+                closeable.close()
+            } catch (ignored: IOException) {
+            }
+        }
+    }
+    // endregion
+    // region: Inner class: BitmapSampled
+    /**
+     * Holds bitmap instance and the sample size that the bitmap was loaded/cropped with.
+     */
+    internal class BitmapSampled(
+        /**
+         * The bitmap instance
+         */
+        val bitmap: Bitmap?,
+        /**
+         * The sample size used to lower the size of the bitmap (1,2,4,8,...)
+         */
+        val sampleSize: Int
+    )
+    // endregion
+    // region: Inner class: RotateBitmapResult
```

Comment:
Please remove all this `region` comments ;) 

Decide to move away from it and bring a better arch in the future

Reply:
Done 😅